### ISSUE TITLE
[java] 7.0.0-rc4 ExceptionAsFlowControl false-positive on Lambda/asynchronous

### ISSUE THREAD

**Issue by blacelle** [pmd/pmd#4815]
**Affects PMD Version:** 
7.0.0-rc4

**Rule:**

ExceptionAsFlowControl
https://pmd.github.io/pmd/pmd_rules_java_design.html#exceptionasflowcontrol

**Description:**
Given rules is a false-positive as the exception in the Lambda would typically not be caught in the outer `catch` block.

**Code Sample demonstrating the issue:**

```java
	protected void doPublish(String issueLabel, Map<?, ?> details) {
	        ExecutorService es = ...;
		try {
			es.execute(() -> {
				try {
					unsafeDoPublish(details);
				} catch (IOException e) {
					throw new UncheckedIOException(e);
				}
			});
		} catch (RuntimeException e) {
			// Do not re-throw else it would go directly to the UncaughtExceptionHandler
			LOGGER.warn("Issue publishing notification " + details, e);
		}
	}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven
